main.py

```diff
@@ -195,24 +195,19 @@ def do_match(cls, flow, args):
             return False
         for name in flow['flow']['match']:
             field_flow = flow['flow']['match'][name]
+            field = args.get(name)
             if name == 'dl_vlan':
-                field = args.get(name)
-                if field:
-                    field = field[-1]
                 if not match_field_dl_vlan(field, field_flow):
                     return False
                 continue
-            if name not in args:
+            if not field:
                 return False
-            field = args[name]
-            if name not in ('ipv4_src', 'ipv4_dst', 'ipv6_src', 'ipv6_dst'):
-                if field_flow != field:
-                    return False
-            else:
-                packet_ip = int(ipaddress.ip_address(field))
-                ip_addr = flow['flow']['match'][name]
-                if packet_ip & ip_addr.netmask != ip_addr.address:
+            if name in ('ipv6_src', 'ipv6_dst'):
+                if not match_field_ip(field, field_flow):
```

Comment:
The `ip` field is now treated with `ip_network` instead of `ip_address`, this allows to use the features of the `ipaddress` library for matching, considering masks. Note that in cases where the mask is not explicitly set, the [default is assumed.](https://github.com/kytos-ng/sdntrace_cp/blob/684db538fb49e264187c18c291e2e86a478ee904/tests/unit/test_utils.py#L424-L468)

Reply:
Nice to have `ip_network` with mask/32 in the input. 

Looks good but the match implementation isn't correctly covering some cases yet, check out my new comment.